typeinfo.go

```diff
@@ -237,7 +237,7 @@ func decodeStructValue(d *Decoder, v reflect.Value) error {
 //------------------------------------------------------------------------------
 
 func encodeCustomValue(e *Encoder, v reflect.Value) error {
-	if v.IsNil() {
+	if v.Kind() == reflect.Ptr && v.IsNil() {
```

Comment:
Most likely this should be extended for array and slice too, but I don't want to do it without test...